### PR TITLE
#143 MWE metric has impediments

### PR DESCRIPTION
**Pull request by uurl**
**Issue #143**

MWE metric has impediments. Need to wait for resolving these impediments, add MWE to reports list and cover it by tests.

### CODE REVIEW

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) please, pay attention to this pull request

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) everybody who has role `REV` is banned at [#247](https://github.com/yegor256/jpeek/issues/247); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by 0crat** [comment]
This pull request [#247](https://github.com/yegor256/jpeek/issues/247) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/C7JGJ00DP/1480782452); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewercom.zerocracy.Par@782f73e4

**Comment by uurl** [comment]
@paulodamaso I don't understand your request.

**Comment by paulodamaso** [comment]
@uurl Don't worry, it was about issue description, according to policy:

> Pull request description explains the solution proposed and contains a link to the original ticket it is related to.

The PR description wasn't addressing the issue, but now that you've corrected it, it's ok. Just enforcing the policy ;)

File: src/test/java/org/jpeek/MetricsTest.java [hunk 1]
```diff
@@ -160,17 +160,17 @@
...
-            new Object[] {"MethodMethodCalls", "LCC", 0.1d},
-            new Object[] {"MethodsWithDiffParamTypes", "LCC", 0.2d},
-            new Object[] {"NoMethods", "LCC", 0.0d},
-            new Object[] {"OneMethodCreatesLambda", "LCC", 0.0d},
-            new Object[] {"OneVoidMethodWithoutParams", "LCC", 0.0d},
-            new Object[] {"OnlyOneMethodWithParams", "LCC", 0.0d},
-            new Object[] {"OverloadMethods", "LCC", 1.0d},
-            new Object[] {"TwoCommonAttributes", "LCC", 0.0d},
-            new Object[] {"WithoutAttributes", "LCC", 0.0d}
+            new Object[] {"Bar", "MWE", 1.0d},
```
Comment (paulodamaso) [inline src/test/java/org/jpeek/MetricsTest.java]: @uurl Why removing LCC metrics?

Comment (uurl) [reply src/test/java/org/jpeek/MetricsTest.java]: @paulodamaso By two reasons:
1. When I did this PR, the LCC metrics test was not implemented yet.
2. There is a limit in the method size, if it is longer than 100 LOC, TravisCI marks an error, so I opened this issue #233
I we just add more lines to that method it violates TravisCI code rules.

Comment (paulodamaso) [reply src/test/java/org/jpeek/MetricsTest.java]: @uurl Right, you should do this way then: merge your changes keeping LCC metrics and put a comment for static analisys tools ignore LOC count check on this class.

Comment (uurl) [reply src/test/java/org/jpeek/MetricsTest.java]: @paulodamaso sorry, I don't know how to do that

Comment (paulodamaso) [reply src/test/java/org/jpeek/MetricsTest.java]: @uurl The tags for ignoring some specific validations are: `@checkstyle` and  `@SupressWarnings("PMD.SomeCheck")` ; there's some examples in jpeek code database.

**Comment by uurl** [comment]
@paulodamaso ready, added the Metrics Tests without overwrite existing

**Comment by paulodamaso** [comment]
@rultor please merge

**Comment by 0crat** [comment]
@paulodamaso[/z](https://www.0crat.com/u/paulodamaso) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by paulodamaso** [comment]
@0crat wait for merge

**Comment by 0crat** [comment]
> @0crat wait for merge ([here](https://github.com/yegor256/jpeek/issues/247#issuecomment-394114519))

@paulodamaso It's a code review job [#247](https://github.com/yegor256/jpeek/issues/247), can't put it on hold

**Comment by uurl** [comment]
@yegor256 It passed the REV. What happen?

**Comment by 0crat** [comment]
The user @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) resigned from [#247](https://github.com/yegor256/jpeek/issues/247), please stop working. Reason for job resignation: It is older than 10 days, see [§8](http://www.zerocracy.com/policy.html#8)

**Comment by 0crat** [comment]
Resigned on delay, see [§8](http://www.zerocracy.com/policy.html#8): -15 point(s) just awarded to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso)

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) everybody who has role `REV` is banned at [#247](https://github.com/yegor256/jpeek/issues/247); I won't be able to assign anyone automatically; consider assigning someone manually (as in [§19](http://www.zerocracy.com/policy.html#19)), or invite more people (as in [§51](http://www.zerocracy.com/policy.html#51)), or remove the job from the scope (as in [§14](http://www.zerocracy.com/policy.html#14))

**Comment by paulodamaso** [comment]
@yegor256 Same thing as https://github.com/yegor256/jpeek/pull/245#issuecomment-395969757, this issue was reviewed a long time ago, so my job was done here.

**Comment by uurl** [comment]
@paulodamaso I guess you must say "quality good"

**Comment by paulodamaso** [comment]
@uurl No, my job was '`REV`, so I just make the review and say it's ok for merge.

**Comment by uurl** [comment]
@rultor merge

**Comment by paulodamaso** [comment]
@yegor256 ping

**Comment by yegor256** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job `gh:yegor256/jpeek#247` is not assigned, can't get performer

**Comment by 0crat** [comment]
The job [#247](https://github.com/yegor256/jpeek/issues/247) is now out of scope

**Comment by paulodamaso** [comment]
@yegor256 I didn't received payment for this job yet; I've made the code revision and asked for merge 7 days before expiring date but got kicked anyways. Since 0crat kicked me when my job was already done it won't pay me ever, it'll be done manually.

**Comment by paulodamaso** [comment]
@yegor256 ping

**Comment by 0crat** [comment]
> @0crat extra @paulodamaso 15 ([here](https://github.com/yegor256/jpeek/issues/247#issuecomment-401627264))

@yegor256 Argument "@paulodamaso" doesn't match regex `\d+min`

**Comment by paulodamaso** [comment]
@yegor256 Didn't worked, please take a look at this

**Pull request merged**